notebooks/tfx_pipelines/pipeline/solutions/pipeline_vertex/model.py

```diff
@@ -0,0 +1,313 @@
+        """Returns the output to be used in the serving signature."""
+        feature_spec = tf_transform_output.raw_feature_spec()
+        feature_spec.pop(features.LABEL_KEY)
+        parsed_features = tf.io.parse_example(
+            serialized_tf_examples, feature_spec
+        )
+
+        transformed_features = model.tft_layer(parsed_features)
+
+        return model(transformed_features)
+
+    return serve_tf_examples_fn
+
+
+def _input_fn(
+    file_pattern: List[str],
+    data_accessor: DataAccessor,
+    tf_transform_output: tft.TFTransformOutput,
+    batch_size: int = 200,
+) -> tf.data.Dataset:
+    """Generates features and label for tuning/training.
+
+    Args:
+      file_pattern: List of paths or patterns of input tfrecord files.
+      data_accessor: DataAccessor for converting input to RecordBatch.
+      tf_transform_output: A TFTransformOutput.
+      batch_size: representing the number of consecutive elements of returned
+        dataset to combine in a single batch
+
+    Returns:
+      A dataset that contains (features, indices) tuple where features is a
+        dictionary of Tensors, and indices is a single Tensor of label indices.
+    """
+    dataset = data_accessor.tf_dataset_factory(
+        file_pattern,
+        dataset_options.TensorFlowDatasetOptions(
+            batch_size=batch_size,
+            label_key=features.transformed_name(features.LABEL_KEY),
+        ),
+        tf_transform_output.transformed_metadata.schema,
+    )
+
+    return dataset
+
+
+def _get_hyperparameters() -> kerastuner.HyperParameters:
+    """Returns hyperparameters for building Keras model."""
+    hp = kerastuner.HyperParameters()
+    # Defines search space.
+    hp.Choice("learning_rate", [1e-2, 1e-3, 1e-4], default=1e-3)
+    hp.Int("n_layers", 1, 2, default=1)
+    with hp.conditional_scope("n_layers", 1):
+        hp.Int("n_units_1", min_value=8, max_value=128, step=8, default=8)
+    with hp.conditional_scope("n_layers", 2):
+        hp.Int("n_units_1", min_value=8, max_value=128, step=8, default=8)
+        hp.Int("n_units_2", min_value=8, max_value=128, step=8, default=8)
+
+    return hp
+
+
+def _build_keras_model(
+    hparams: kerastuner.HyperParameters,
+    tf_transform_output: tft.TFTransformOutput,
+) -> tf.keras.Model:
+    """Creates a Keras WideDeep Classifier model.
+    Args:
+      hparams: Holds HyperParameters for tuning.
+      tf_transform_output: A TFTransformOutput.
+    Returns:
+      A keras Model.
+    """
+    deep_columns = [
+        tf.feature_column.numeric_column(
+            key=features.transformed_name(key), shape=()
+        )
+        for key in features.NUMERIC_FEATURE_KEYS
+    ]
+
+    input_layers = {
+        column.key: tf.keras.layers.Input(
+            name=column.key, shape=(), dtype=tf.float32
+        )
+        for column in deep_columns
+    }
+
+    categorical_columns = [
+        tf.feature_column.categorical_column_with_identity(
```

Comment:
No action, just wanted to flag these to be updated to preprocessing layers since this is TF2.6 on subsequent update.